build.rs

```diff
@@ -213,3 +220,22 @@ fn set_cfg(cfg: &str) {
     }
     println!("cargo:rustc-cfg={}", cfg);
 }
+
+fn is_musl_time64_abi() -> bool {
+    match env::var("TARGET") {
+        Ok(target) => match &target[..] {
+            "arm-unknown-linux-musleabi"
+            | "arm-unknown-linux-musleabihf"
+            | "armv5te-unknown-linux-musleabi"
+            | "armv7-unknown-linux-musleabi"
+            | "armv7-unknown-linux-musleabihf"
+            | "i586-unknown-linux-musl"
+            | "i686-unknown-linux-musl"
+            | "mips-unknown-linux-musl"
+            | "mipsel-unknown-linux-musl"
+            | "powerpc-unknown-linux-musl" => true,
+            _ => false,
+        },
+        Err(_) => false,
+    }
+}
```

Comment:
We probably shouldn't match literal targets here since we only care about the prefix and the suffix. Instead could we:

- Add `const TARGET_PREFIXES_32BIT: &[&str] = &[/* ... */];` that contains `arm-`, `i586-`, etc
- Add `const MUSL_SUFFIXES : &[&str] = &[/* ... */];` with `-musleabi`, `-musl`, etc
- Add `fn target_is_32bit(target: &str) -> bool` that checks if it starts if `target` starts with any of `TARGET_PREFIXES_32BIT`. Then we can reuse it with the GNU change.
- Add `fn target_is_musl(target: &str) -> bool` that does the same with the suffixes
- Add `fn target_is_linux(target: &str) -> bool` that checks if it contains `-linux-`
- Check these three functions in order to set `musl_redir_time64`/`musl_time64_abi`